src/osal/linux/osal.c

```diff
@@ -568,12 +569,61 @@ uint8_t os_buf_header(os_buf_t *p, int16_t header_size_increment)
    return 255;
 }
 
+/**
+ * Convert IPv4 address to string
+ * @param ip               In: IP address
+ * @param outputstring     Out: Outputstring. Should have length INET_ADDRSTRLEN.
+ */
+void ip_to_string(
+   uint32_t                ip,
+   char                    *outputstring)
+{
+   uint32_t                ip_network_endianness = htonl(ip);
```

Comment:
I wonder if this would be correct on a big-endian linux? I.e. is the linux representation of an ip address as a 32-bit integer dependent on the endianness of the machine? Not sure. However there are probably many other instances where BE would not work anyway.

Alternatively provide an inverse of the OS_IP4_ADDR_TO_U32 macro and use that?

Reply:
We need `htonl()` for `inet_ntop()` to work. On a big-endian machine `htonl()` would be a no-op, I guess.

It is done pretty much the same for rt-kernel: https://github.com/rtlabs-com/p-net/blob/4a960c8725387a90e511e0975021860d5486f1f2/src/osal/rt-kernel/osal.c#L54

Reply:
Yes, htonl is needed for a little-endian machine. Whether it is needed for a big-endian machine I'm not sure, it would depend on the internal linux representation of an IP address as a 32-bit integer on a BE target.

As for rt-kernel (LWIP) at a quick check it seems that IP addresses are always in network format, so it would probably be correct there.

However I think this is ok for now but for pnet we should perhaps consider using the platform (network stack) representation of IP address rather than inventing our own and having to convert between representations.